src/ops.rs

```diff
@@ -79,3 +82,43 @@ impl<A: BitSetLike> BitSetLike for BitSetNot<A> {
         !self.0.layer0(i)
     }
 }
+
+macro_rules! operator {
+    ( $bitset:ident [ $( $arg:ident ),* ] ) => {
+        impl<$( $arg ),*> Not for $bitset<$( $arg ),*>
+            where $( $arg: BitSetLike ),*
+        {
+            type Output = BitSetNot<Self>;
+            fn not(self) -> Self::Output {
+                BitSetNot(self)
+            }
+        }
+
+        impl<$( $arg,  )* T> BitAnd<T> for $bitset<$( $arg ),*>
+            where T: BitSetLike,
+                  $( $arg: BitSetLike ),*
+        {
+            type Output = BitSetAnd<Self, T>;
+            fn bitand(self, rhs: T) -> Self::Output {
+                BitSetAnd(self, rhs)
+            }
+        }
+
+        impl<$( $arg, )* T> BitOr<T> for $bitset<$( $arg ),*>
+            where T: BitSetLike,
+                  $( $arg: BitSetLike ),*
+        {
+            type Output = BitSetOr<Self, T>;
+            fn bitor(self, rhs: T) -> Self::Output {
+                BitSetOr(self, rhs)
+            }
+        }
+
+    }
+}
+
+operator!(BitSet [ ]);
+operator!(BitSetAnd [ A, B ]);
+operator!(BitSetNot [ A ]);
+operator!(BitSetOr [ A, B ]);
```

Comment:
My experience is that macro invocations with [] are touched by rustfmt.

Reply:
I replaced them with ( ) since running rustfmt gave me some weird results (formatting on some of them and leaving some the same).